src/com/etendoerp/dependencymanager/util/PackageUtil.java

```diff
@@ -221,6 +224,56 @@ public static synchronized void updateOrCreateDependency(String group, String ar
     OBDal.getInstance().save(existingDependency);
   }
 
+  /**
+   * Fetches the latest package version from the database using OBCriteria.
+   *
+   * @param group The group of the package.
+   * @param artifact The artifact of the package.
+   * @return The latest version string or null if no versions are found.
+   */
+  private static String getLastPackageVersionFromDB(String group, String artifact) {
+    OBCriteria<PackageVersion> criteria = OBDal.getInstance().createCriteria(PackageVersion.class);
+    criteria.createAlias("package", "pkg");
+    criteria.add(Restrictions.eq("pkg.artifact", artifact));
+    criteria.add(Restrictions.eq("pkg.group", group));
+    criteria.addOrder(Order.desc(VERSION));
+    criteria.setMaxResults(1);
+
+    List<PackageVersion> packageVersions = criteria.list();
+    if (!packageVersions.isEmpty()) {
+      return packageVersions.get(0).getVersion();
+    }
+    return null;
+  }
+
+  /**
+   * Splits the provided version range string into a two-element array.
+   *
+   * @param versionRange The version range string to split.
+   * @return A two-element array with the start and end versions.
+   */
+  public static String[] splitCoreVersionRange(String versionRange) {
+    String cleanedRange = versionRange.replaceAll("[\\[\\]()]", "");
+    String[] versionSplit = cleanedRange.split(",");
+    if (versionSplit.length != 2) throw new IllegalArgumentException(OBMessageUtils.messageBD("ETDEP_Invalid_Version_Range_Format") + versionRange);
```

Comment:
its more safe the use of {} for the if then block

```suggestion
    if (versionSplit.length != 2){
     throw new IllegalArgumentException(OBMessageUtils.messageBD("ETDEP_Invalid_Version_Range_Format") + versionRange);

    }
```